fvm/src/trace/mod.rs

```diff
@@ -26,6 +30,57 @@ pub enum ExecutionEvent {
         params: Option<IpldBlock>,
         value: TokenAmount,
     },
+    SpanBegin(SpanBegin),
+    SpanEnd(SpanEnd),
     CallReturn(ExitCode, Option<IpldBlock>),
     CallError(SyscallError),
 }
+
+#[derive(Clone, Debug)]
+pub struct SpanBegin {
+    /// User-supplied label for this span.
+    pub label: String,
+    /// User-supplied tag for this span.
+    pub tag: String,
+    /// Parent span.
+    pub parent: SpanId,
+    /// CID of the currently executing method's code.
+    pub code: Cid,
+    /// Number of the currently executing method.
+    pub method: MethodNum,
+    /// The timestamp when this event ocurred, in nanoseconds.
+    pub timestamp: u64,
+}
+
+#[derive(Clone, Debug)]
+pub struct SpanEnd {
+    /// The ID of the span that is ending.
+    pub id: SpanId,
+    /// The timestamp when this event ocurred, in nanoseconds.
+    pub timestamp: u64,
```

Comment:
To make sure I understand correctly: would you like to do this instead of the span functionality, or would you like to do this as well as the span functionality?

I believe `InvokeActor` and `ExitActor` are equivalent to what we have today with `Call`, `CallReturn`, and `CallError`.

Just for the record, I think that it would be useful to include the timestamp on every event. I think a change like this deserves a separate issue/PR though.

Reply:
Also, regarding your previous comment:
> For the code CID, I'm adding a special event when we invoke an actor so we shouldn't need it here (PR WIP).

are you already adding `InvokeActor`?

Reply:
> To make sure I understand correctly: would you like to do this instead of the span functionality, or would you like to do this as well as the span functionality?

In addition to spans. The idea is to split "spans" from "timestamps".

But you're right, doing it in a second PR makes sense.

Reply:
> are you already adding InvokeActor?

@fridrik01 is likely going to do it in a few days as he'll need it for some tracing work he's doing in lotus.